nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/utils/AzureStorageUtils.java

```diff
@@ -146,24 +142,46 @@ private AzureStorageUtils() {
     }
 
     /**
-     * Create CloudBlobClient instance.
+     * Create BlobServiceClient instance.
      * @param flowFile An incoming FlowFile can be used for NiFi Expression Language evaluation to derive
      *                 Account Name, Account Key or SAS Token. This can be null if not available.
      */
-    public static CloudBlobClient createCloudBlobClient(ProcessContext context, ComponentLog logger, FlowFile flowFile) throws URISyntaxException {
+    public static BlobServiceClient createBlobServiceClient(PropertyContext context, FlowFile flowFile) {
```

Comment:
This will break things for any downstream `AzureStorageUtils` users.

Reply:
@jfrazee I don't think we can achieve this type of downstream compatability without having two versions of the azure-storage processors available, one for the legacy sdk and one for the v12 sdk. 

What would your suggestion be?

Reply:
I made an overall comment on the PR. We wouldn't necessarily have to have two sets of processors, but let's say we say we're just going to break it, I think now would be a good time to improve things by wrapping it in something we control.

Reply:
@jfrazee thank you for your comments the other day. Since Microsoft/Azure SDK's ServiceClient are not implements any service, I've created an custom AzureServiceClient abstract class that will be extended by AzureBlob / AzureQueue ServiceClient abstract classes, in which we wrap the functions for this service client. Although I feel this is only a small level of control - from here onwards we could include required functions from these wrapper objects.

What do you think about this solution?